docs: adding HIL sanity badges

milapjoshi opened the pull request:
<!--
    Tag your PR title with the components that it touches along with
    the type of change
    E.g. "fix(orc8r): Fix reindexer race condition" or "feat(agw) ..."
-->

## Summary

- one for last stable and one for current state 
- removing PR welcome badge as discussed with brian/tariq

We are running nightly HIL testing. so using these badge, we want to achive two things. 
Indicate last HIL sanity stable build, which can be used to create quick release branch.
Indicate latest hil testing status.

## Test Plan

<!--
    How did you test your change? How do you know it works?
    Add supporting screenshots, terminal pastes, etc. as necessary
-->

## Additional Information

- [ ] This change is backwards-breaking

<!--
    If this is a backwards-breaking change, document the upgrade instructions.
    All upgrade instructions for backwards-breaking changes will be aggregated
    in the next release's changelog so this is very important to fill out.
-->

milap-home committed:
adding HIL sanity badge one for last stable and one for current state; removing PR welcome badge as disccussed with brian

Signed-off-by: Milap Joshi <milapjoshi@fb.com>

Signed-off-by: milap joshi <milap.joshi@gmail.com>

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milapjoshi <milap.joshi@gmail.com>

milap-home committed:
adding HIL sanity badge one for last stable and one for current state; removing PR welcome badge as disccussed with brian

Signed-off-by: Milap Joshi <milapjoshi@fb.com>

Signed-off-by: milap joshi <milap.joshi@gmail.com>

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milapjoshi <milap.joshi@gmail.com>

milap-home committed:
adding HIL sanity badge one for last stable and one for current state; removing PR welcome badge as disccussed with brian

Signed-off-by: Milap Joshi <milapjoshi@fb.com>

Signed-off-by: milap joshi <milap.joshi@gmail.com>

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milapjoshi <milap.joshi@gmail.com>

milap-home committed:
adding HIL sanity badge one for last stable and one for current state; removing PR welcome badge as disccussed with brian

Signed-off-by: milap joshi <milap.joshi@gmail.com>

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milapjoshi <milap.joshi@gmail.com>

milap-home committed:
signing off

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milap joshi <milap.joshi@gmail.com>

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milapjoshi <milap.joshi@gmail.com>

milap-home committed:
signing off

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milap joshi <milap.joshi@gmail.com>

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milapjoshi <milap.joshi@gmail.com>

milap-home committed:
Merge branch 'hil_badge' of https://github.com/magma/magma into hil_badge

Signed-off-by: milap joshi <milap.joshi@gmail.com>

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milapjoshi <milap.joshi@gmail.com>

hcgatewood reviewed (changes requested):
This feels like a weird addition, or at least one that's eng-driven rather than user-driven. The top of a readme is meant to communicate information to top-of-funnel

1. The badge text doesn't convey semantically meaningful information to new users -- what does "AGW HIL sanity" mean to a common user? Is this the latest "commit that's known stable via basic HIL testing", or ?
2. The badge link takes you to a (FB-internal ?) Grafana login page, and I have no idea how to log in (I'm logged in to my personal grafana, but that seems ineffective)
3. If you want to remove the "PRs welcome" badge, please provide explicit rationale

For those reasons, I think this type of badge is a better fit for the Docusaurus, which targets middle or bottom of funnel, rather than the root readme, which targets top of funnel. You can also place context and directives for how to use the badge if we go with the Docusaurus approach

milapjoshi commented:
> This feels like a weird addition, or at least one that's eng-driven rather than user-driven. The top of a readme is meant to communicate information to top-of-funnel
> 
> 1. The badge text doesn't convey semantically meaningful information to new users -- what does "AGW HIL sanity" mean to a common user? Is this the latest "commit that's known stable via basic HIL testing", or ?
mj>>  "Is this the latest "commit that's known stable via basic HIL testing", or ? " >> yes, we can rephrase or provide link in docusaurus to provide more description. basically, we want to make community aware about 'Hardware in loop' testing done on magma main line build.
> 2. The badge link takes you to a (FB-internal ?) Grafana login page, and I have no idea how to log in (I'm logged in to my personal grafana, but that seems ineffective)
mj>> badge link is open to public. we can provide credential for any way to access it. like circleci access control. (As of now its static shared credential)
> 3. If you want to remove the "PRs welcome" badge, please provide explicit rationale
MJ>> This came from brian. Rational is -  badge on root READme file is prime real estate. So, if PR welcome is not changing to any other message during merge freeze it can be removed to make space.
> 
> For those reasons, I think this type of badge is a better fit for the Docusaurus, which targets middle or bottom of funnel, rather than the root readme, which targets top of funnel. You can also place context and directives for how to use the badge if we go with the Docusaurus approach

MJ>> ok let me take this action items :
   - work with Jerad to find proper place in docusaurus to provide more information about HW in loop testing.
   - one badge of latest HIL sanity run (pass/fail) can be still on root readme with link to docusaurus 
   - one badge about last HIL sanity passed build would be place on internal docusaurus page.

milapjoshi committed:
added hil test details

Signed-off-by: Milap Joshi <milapjoshi@fb.com>

Signed-off-by: milap joshi <milap.joshi@gmail.com>

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milapjoshi <milap.joshi@gmail.com>

milap-home committed:
added hil tests details

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milap joshi <milap.joshi@gmail.com>

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milapjoshi <milap.joshi@gmail.com>

milapjoshi committed:
added badge in hil readme and update main line readme badge

Signed-off-by: Milap Joshi <milapjoshi@fb.com>

Signed-off-by: milap joshi <milap.joshi@gmail.com>

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milapjoshi <milap.joshi@gmail.com>

milapjoshi committed:
added links

Signed-off-by: Milap Joshi <milapjoshi@fb.com>

Signed-off-by: milap joshi <milap.joshi@gmail.com>

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milapjoshi <milap.joshi@gmail.com>

milapjoshi committed:
added links

Signed-off-by: Milap Joshi <milapjoshi@fb.com>

Signed-off-by: milap joshi <milap.joshi@gmail.com>

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milapjoshi <milap.joshi@gmail.com>

milapjoshi committed:
lint check

Signed-off-by: Milap Joshi <milapjoshi@fb.com>

Signed-off-by: milap joshi <milap.joshi@gmail.com>

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milapjoshi <milap.joshi@gmail.com>

milapjoshi committed:
added links

Signed-off-by: Milap Joshi <milapjoshi@fb.com>

Signed-off-by: milap joshi <milap.joshi@gmail.com>

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milapjoshi <milap.joshi@gmail.com>

milapjoshi committed:
readme lint checks

Signed-off-by: Milap Joshi <milapjoshi@fb.com>

Signed-off-by: milap joshi <milap.joshi@gmail.com>

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milapjoshi <milap.joshi@gmail.com>

milapjoshi committed:
Merge branch 'hil_badge' of https://github.com/magma/magma into hil_badge

* 'hil_badge' of https://github.com/magma/magma:

Signed-off-by: milap joshi <milap.joshi@gmail.com>

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milapjoshi <milap.joshi@gmail.com>

milapjoshi committed:
readme h3 to h2

Signed-off-by: Milap Joshi <milapjoshi@fb.com>

Signed-off-by: milap joshi <milap.joshi@gmail.com>

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milapjoshi <milap.joshi@gmail.com>

milap-home committed:
sign off from worker node

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milap joshi <milap.joshi@gmail.com>

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milapjoshi <milap.joshi@gmail.com>

hcgatewood reviewed (commented):
Can you give screenshots of what these changes look like 🙂

milapjoshi commented:
@hcgatewood  you think this approach is ok. i have readme file with details of what HIL is and badge is moved inside that readme file. however, we still have top of the funnel badge to indicate hil sanity status. with link to internal readme page.

milapjoshi committed:
point to last stable badge here
Signed-off-by: milap joshi <milap.joshi@gmail.com>

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milapjoshi <milap.joshi@gmail.com>

milapjoshi committed:
updated test cases description
Signed-off-by: milap joshi <milap.joshi@gmail.com>

Signed-off-by: milap joshi <milapjoshi@gmail.com>

Signed-off-by: milapjoshi <milap.joshi@gmail.com>

milapjoshi commented:
> Can you give screenshots of what these changes look like 🙂

![image](https://user-images.githubusercontent.com/5917909/133149788-481ae1cb-d961-4a5b-8dc5-4f62d031b793.png)

![image](https://user-images.githubusercontent.com/5917909/133150530-7e4e024b-6da9-4bc5-939c-9664de1fa95b.png)

https://github.com/magma/magma/blob/hil_badge/docs/readmes/lte/hil_tests.md

milapjoshi commented:
@hcgatewood can you please check on this.

hcgatewood reviewed (changes requested):
This looks reasonable. Couple notes

- Spelling/grammar
- Dashboard still isn't accessible by me directly, and there are no instructions on how to access it
- The word "sanity" can be removed everywhere -- just call it "HIL AGW tests" or similar

milapjoshi committed:
removed sanity word and updated dashboard access instruction

Signed-off-by: milap joshi <milap.joshi@gmail.com>

milapjoshi commented:
> This looks reasonable. Couple notes
> 
> * Spelling/grammar
> * Dashboard still isn't accessible by me directly, and there are no instructions on how to access it
> * The word "sanity" can be removed everywhere -- just call it "HIL AGW tests" or similar
@hcgatewood 
--removed sanity and corrected some typo. (Badges would update with nightly run, and would come as `HIL AGW tests`
dashboard should be accessible. i added some credential details. its view only cred.
you can check here - https://github.com/magma/magma/tree/hil_badge

hcgatewood commented on docs/readmes/lte/hil_tests.md:
```diff
@@ -0,0 +1,48 @@
+### Dashboard
+
+All test results are available to compare on [dashboard](http://automation.fbmagma.ninja/). Please use `username:magma` and `password:magma`.
+We can retrieve log and Grafana metrics for each run by clicking on test run result table.
```
Go ahead and move this to the top of the doc, since it's probably the most important link

hcgatewood reviewed (approved):
Awesome! I was able to take a look at the dashboard. Thanks for adding this!

milapjoshi committed:
move dashboard section to top of readme

Signed-off-by: milap joshi <milap.joshi@gmail.com>

milapjoshi committed:
move badge link under top level heading

Signed-off-by: milap joshi <milap.joshi@gmail.com>

milapjoshi committed:
put git badge with para

Signed-off-by: milap joshi <milap.joshi@gmail.com>

milapjoshi committed:
trying linter error solution

Signed-off-by: milap joshi <milap.joshi@gmail.com>

milapjoshi committed:
trying linter error

Signed-off-by: milap joshi <milap.joshi@gmail.com>

milapjoshi committed:
local check passing for lint

Signed-off-by: milap joshi <milap.joshi@gmail.com>

milapjoshi committed:
reverify check

Signed-off-by: milap joshi <milap.joshi@gmail.com>
